lib/cocoapods/validator.rb

```diff
@@ -28,8 +28,12 @@ class Validator
     #         the Source URLs to use in creating a {Podfile}.
     #
     def initialize(spec_or_path, source_urls)
-      @source_urls = source_urls.map { |url| config.sources_manager.source_with_name_or_url(url) }.map(&:url)
       @linter = Specification::Linter.new(spec_or_path)
+      @source_urls = if !@linter.spec || @linter.spec.dependencies.empty?
```

Comment:
This doesn't seem like the right fix to me because we're emptying out the `source_urls` parameter with the goal of short circuiting a `pod repo update` in the future knowing that it can't do anything if there are no repos to update. What if we just set `repo_update = false` on `@installer` in `download_pod` if we know there are no dependencies (and this is `pod lib lint` and not `pod spec lint`? What do you think?